OpenGLUnitTests/GeometryTests.cs

```diff
@@ -0,0 +1,139 @@
+
+namespace OpenGLUnitTests
+{
+    [TestClass]
+    public class GeometryTests
+    {
+        [TestMethod]
+        public void CalculateNormalsSingleTriangle()
+        {
+            Vector3[] vertices = new Vector3[]
+            {
+                new Vector3(0, 1, 0),
+                new Vector3(1, 0, 0),
+                new Vector3(0, 0, 0)
+            };
+            uint[] elements = new uint[] { 0, 1, 2 };
+
+            Vector3[] expectedNormals = new Vector3[]
+            {
+                new Vector3(0, 0, -1),
+                new Vector3(0, 0, -1),
+                new Vector3(0, 0, -1)
+            };
+
+            VerifyOverloads(vertices, elements, expectedNormals);
+        }
+
+        [TestMethod]
+        public void CalculateNormalsSingleTriangleReversed()
+        {
+            Vector3[] vertices = new Vector3[]
+            {
+                new Vector3(0, 1, 0),
+                new Vector3(1, 0, 0),
+                new Vector3(0, 0, 0)
+            };
+            uint[] elements = new uint[] { 2, 1, 0 };
+
+            Vector3[] expectedNormals = new Vector3[]
+            {
+                new Vector3(0, 0, 1),
+                new Vector3(0, 0, 1),
+                new Vector3(0, 0, 1)
+            };
+
+            VerifyOverloads(vertices, elements, expectedNormals);
+        }
+
+        [TestMethod]
+        public void CalculateNormalsMultipleTriangles()
+        {
+            // Pyramid shape
+            Vector3[] vertices = new Vector3[]
+            {
+                new Vector3( 0, 1,  0),
+                new Vector3( 0, 0,  1),
+                new Vector3( 1, 0,  0),
+                new Vector3( 0, 0, -1),
+                new Vector3(-1, 0,  0)
+            };
+            uint[] elements = new uint[] 
+            { 
+                0, 1, 2,
+                0, 2, 3,
+                0, 3, 4,
+                0, 4, 1
+            };
+
+            Vector3[] expectedNormals = new Vector3[]
+            {
+                new Vector3( 0, 1,  0).Normalize(),
+                new Vector3( 0, 1,  1).Normalize(),
+                new Vector3( 1, 1,  0).Normalize(),
+                new Vector3( 0, 1, -1).Normalize(),
+                new Vector3(-1, 1,  0).Normalize()
+            };
+
+            VerifyOverloads(vertices, elements, expectedNormals);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentOutOfRangeException))]
+        public void CalculateNormalWrongSizeNormalArray()
+        {
+            Geometry.CalculateNormals(new Vector3[3], new uint[3], new Vector3[2]);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentOutOfRangeException))]
+        public void CalculateNormalWrongSizeElementArray()
+        {
+            Geometry.CalculateNormals(new Vector3[3], new uint[4]);
+        }
+
+        private void VerifyOverloads(Vector3[] vertices, uint[] elements, Vector3[] expectedNormals)
+        {
+            {
```

Comment:
These brackets are here to provide visual separation between independent tests (and re-use of variable names I suppose)?